test/evp_extra_test.c

```diff
@@ -4647,6 +4657,80 @@ const OPTIONS *test_get_options(void)
     return options;
 }
 
+/* Test that trying to sign with a public key errors out gracefully */
+static int test_ecx_not_private_key(int tst)
+{
+#ifdef OPENSSL_NO_EC
+    return 1;
+#else
```

Comment:
I'd prefer if the whole function was in `#ifndef OPENSSL_NO_EC` and the ADD_ALL_TESTS was also ifndef-ed.

Reply:
fixed